src/backend/mpi/coherence.cpp

```diff
@@ -7,6 +7,7 @@
 #include "../backend.hpp"
 #include "swdsm.h"
 #include "virtual_memory/virtual_memory.hpp"
+#include "write_buffer.hpp"
```

Comment:
As far as I can tell this header is only used by the `argo_write_buffer` object in this file, which in turn is declared in swdsm.h and defined in swdsm.hpp. Is it appropriate to include the header again here?

Reply:
The block_size constant is also defined in the header, but redefined identically a few times in this file. I think this is an error in the header to export the constant into the global namespace.